### PR TITLE
Refactor runtime adapter abstract fetch

### PR DESCRIPTION
**Pull request by paulomarg**
### WHY are these changes introduced?

In order to properly use the new clients in a runtime-agnostic way, we'll need to pass in a `fetch` implementation when creating them.

While we already have this logic in `shopify-api`, it's currently based on our `NormalizedRequest` and `NormalizedResponse` objects, which are essentially translating `fetch` for every runtime we support.

### WHAT is this pull request doing?

Dropping the middle man here and making `fetch` the API for `abstractFetch`. While this is a breaking change for apps using custom runtimes, it's very likely that any environments are simply doing what we do and translating `fetch` as well. If not, they can be updated to conform with the Web API pattern in the same way they're converted to our own format now.

This has the advantage of bringing us closer to the community standard rather than introducing our own, and I feel it's more likely to make it _easier_ to add new runtimes in the future.

## Type of change

- [x] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have used `yarn changeset` to create a draft changelog entry (do NOT update the `CHANGELOG.md` file manually)
- [x] I have added/updated tests for this change
- [x] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

File: .changeset/rich-ladybugs-own.md
Comment (byrichardpowell) [inline .changeset/rich-ladybugs-own.md:7]: ~~Worth giving a code example that shows what using a default adapter from this package means? Wouldn't suprise me if partners don't understand what this means, might be better to show them?~~

The migration guide is great, is it worth linking to this here?

File: packages/shopify-api/adapters/node/index.ts
Comment (byrichardpowell) [inline packages/shopify-api/adapters/node/index.ts:25]: Is there anyway to improve the type of `AbstractFetchFunc` so that we don't need to cast as any?

**Review by byrichardpowell — APPROVED** [review]
A few small comments, but approving so you don't need a second review.

**Review by lizkenyon — APPROVED** [review]

**Pull request merged**
